File: packages/java-sdk/src/main/java/com/turbodocx/models/RecipientResponse.java
package com.turbodocx.models;

import com.google.gson.annotations.SerializedName;

/**
 * Represents a recipient in API responses
 */
public class RecipientResponse {
    @SerializedName("id")
    private String id;

    @SerializedName("name")
    private String name;

    @SerializedName("email")
    private String email;

    @SerializedName("signUrl")
    private String signUrl;

    @SerializedName("signedAt")
    private String signedAt;

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public String getSignUrl() {
        return signUrl;
    }

    public String getSignedAt() {
        return signedAt;
    }
}

File: packages/php-sdk/src/Types/Responses/RecipientResponse.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Responses;

use TurboDocx\Types\RecipientStatus;

/**
 * Recipient information in document status response
 */
final class RecipientResponse
{
    public function __construct(
        public string $id,
        public string $email,
        public string $name,
        public RecipientStatus $status,
        public ?string $signUrl,
        public ?string $signedAt,
    ) {}

    /**
     * Create from array
     *
     * @param array<string, mixed> $data
     * @return self
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'] ?? '',
            email: $data['email'] ?? '',
            name: $data['name'] ?? '',
            status: RecipientStatus::from($data['status'] ?? 'pending'),
            signUrl: $data['signUrl'] ?? null,
            signedAt: $data['signedAt'] ?? null,
        );
    }
}
